core/src/Dscp/Core/Foundation/Witness.hs

```diff
@@ -270,33 +271,55 @@ data BlockToSign =
     deriving (Eq, Show, Generic)
 
 data Header = Header
-    { hSignature  :: Signature BlockToSign
-    , hIssuer     :: PublicKey
-    , hDifficulty :: Difficulty
-    , hPrevHash   :: HeaderHash
+    { hSignature  :: !(Signature BlockToSign)
+    , hIssuer     :: !PublicKey
+    , hDifficulty :: !Difficulty
+    , hSlotId     :: !SlotId
+    , hPrevHash   :: !HeaderHash
```

Comment:
`{-# LANGUAGE StrictData #-}`

Reply:
It feels better to be explicit somehow. Maybe I'll rethink it later.

Reply:
I find it useful for modules consisting of dumb types like parameters set. Otherwise manual control seems more preceded since making field strict wrongly can cause runtime errors in special cases or degrade performance